context.go

```diff
@@ -66,52 +71,99 @@ type Context struct {
 //
 //	decCtx, err := srtp.CreateContext(key, salt, profile, srtp.SRTPReplayProtection(256))
 func CreateContext(masterKey, masterSalt []byte, profile ProtectionProfile, opts ...ContextOption) (c *Context, err error) {
-	keyLen, err := profile.KeyLen()
+	c = &Context{
+		srtpSSRCStates:  map[uint32]*srtpSSRCState{},
+		srtcpSSRCStates: map[uint32]*srtcpSSRCState{},
+		profile:         profile,
+		mkis:            map[string]srtpCipher{},
+	}
+
+	for _, o := range append(
+		[]ContextOption{ // Default options
+			SRTPNoReplayProtection(),
+			SRTCPNoReplayProtection(),
+		},
+		opts..., // User specified options
+	) {
+		if errOpt := o(c); errOpt != nil {
+			return nil, errOpt
+		}
+	}
+
+	err = c.AddCipherForMKI(c.sendMKI, masterKey, masterSalt)
 	if err != nil {
 		return nil, err
 	}
+	c.cipher = c.mkis[string(c.sendMKI)]
```

Comment:
Is it always safe to use string casted bytes as key? (e.g. against non-printable chars and multi-bytes chars)

Reply:
Yes, it works fine.